docs/concepts.rst

```diff
@@ -27,40 +32,77 @@ Projects
 
```

Comment:
Need to delineate the differences between product, project and package. Also, stress the importance of version control because this above all things is what separates CumulusCI from Salesforce and SalesforceDX, which in turn leads us to emphasize on automation and its importance to this framework (and how it ultimately saves a lot of time for client).